docs/migration.md

```diff
@@ -2,96 +2,253 @@
+| `model_json_schema()` | `json_schema()` |
+| `model_construct()` | `construct()` |
+| `model_validate()` | `parse_obj()` |
+| `__pydantic_validator__` | `__validators__` |
+| `model_fields` | `__fields__` |
+| `__signature__` | `__signature__` |
+| `__class_vars__` | `__class_vars__` |
+| `__pydantic_private__` | `__private_attributes__` |
+
+* Some of the built-in data-loading functionality has been slated for removal. In particular,
+    `parse_raw` and `parse_file` are now deprecated. In Prefect V2, `model_validate_json` works like `parse_raw`. Otherwise, you should load the data and then pass it to `model_validate`.
+* The `from_orm` method has been removed; you can now just use `model_validate` (equivalent to `parse_obj` from
+  Pydantic V1) to achieve something similar, as long as you've set `from_attributes=True` in the model config.
+* The `__eq__` method has changed for models.
+    * Models can only be equal to other `BaseModel` instances.
+    * For two model instances to be equal, they must have the same:
+        * Type (or, in the case of generic models, non-parametrized generic origin type)
+        * Field values
+        * Extra values (only relevant when `model_config['extra'] == 'allow'`)
+        * Private attribute values; models with different values of private attributes are no longer equal.
+        * Models are no longer equal to the dicts containing their data.
+        * Non-generic models of different types are never equal.
+        * Generic models with different origin types are never equal. We don't require *exact* type equality so that,
+            for example, instances of `MyGenericModel[Any]` could be equal to instances of `MyGenericModel[int]`.
+* We have replaced the use of the `__root__` field to specify a "custom root model" with a new type called
+    [`RootModel`](usage/models.md#custom-root-types) which is intended to replace the functionality of using a field
+    called `__root__` in Pydantic V1.
+* We have significantly expanded Pydantic's capabilities related to customizing serialization. In particular, we have
+    added the [`@field_serializer`](api/functional_serializers.md#pydantic.functional_serializers.field_serializer),
+    [`@model_serializer`](api/functional_serializers.md#pydantic.functional_serializers.model_serializer), and
+    [`@computed_field`](api/fields.md#pydantic.fields.computed_field) decorators, which each address various
+    shortcomings from Pydantic V1.
+    * Due to performance overhead and implementation complexity, we have now removed support for specifying
+        `json_encoders` in the model config. This functionality was originally added for the purpose of achieving custom
+        serialization logic, and we think the new serialization decorators are a better choice in most common scenarios.
+        However, if your usage of `json_encoders` is not compatible with the new serialization decorators,
+        please create a GitHub issue letting us know.
```

Comment:
We should link the `Custom Types` page here, or at least mention how to use `__get_pydantic_json_schema_schema__`.